pkg/new-ui/v1beta1/README.md

```diff
@@ -0,0 +1,123 @@
+1. Build and serve only the frontend. The dev server will also be proxying requests to the backend
+2. Build the frontend and serve it via the backend locally
+
+### Serve only the frontend
+
+You can run a webpack dev server that only exposes the frontend files, which can be useful for testing only the UI of the app. There's also a `proxy.conf.json` file which configures the dev server to send the backend requests to port `8000`.
+
+In order to build the UI locally, and expose it with a webpack dev server you will need to:
+1. Create a module from the [common library](https://github.com/kubeflow/kubeflow/tree/master/components/crud-web-apps/common/frontend/kubeflow-common-lib)
+2. Install the node modules of the app and also link the common-library module
+
+
+
+You can build the common library with:
+```bash
+cd /tmp && git clone https://github.com/kubeflow/kubeflow.git \
+  && cd kubeflow \
+  && git checkout a349284 \
+  && cd components/crud-web-apps/common/frontend/kubeflow-common-lib
+
+# build the common library module
+npm i
+npm run build
+
+# link the module to your npm packages
+# depending on where you npm stores the global packages you
+# might need to use sudo
+npm link dist/kubeflow
+```
+
+And then build and run the UI locally, on `localhost:4200`, with:
+```bash
+# If you've already cloned the repo then skip this step and just
+# navigate to the pkg/new-ui/v1beta1/frontend dir
+cd /tmp && git clone https://github.com/kubeflow/katib.git \
+  && cd katib/pkg/new-ui/v1beta1/frontend
+
+npm i
+npm link kubeflow
+npm run start
+```
+
+### Serve the UI  from the backend
+
+This is the recommended way to test the web app e2e. In order to build the UI and serve it via the backend, locally, you will need to:
+
+1. Build the UI locally. You can follow the steps from the previous section, but instead of running `npm run start` you will need to run `npm run build:watch`. This will start a process that will be watching the source code and building the frontend artifacts under `frontend/dist/static` folder.
+1. Run `kubectl port-forward svc/katib-db-manager 6789 -n kubeflow` to expose `katib-db-manager` service for external access. You can use [different ways](https://kubernetes.io/docs/tasks/access-application-cluster/) to get external address for Kubernetes service. After exposing service, you should be able to receive information by running `wget <external-ip>:<service-port>`. In case of port-forwarding above, you have to run `wget localhost:6789`.
+
+1. Go to `cmd/new-ui/v1beta1`.
+
+1. Run `main.go` file with appropriate flags, where:
+
+   - `--build-dir` - directory with the frontend artifacts.
+   - `--port` - port to access Katib UI.
+   - `--db-manager-address` - Katib DB manager external IP and port address.
+
+   For example, if you use port-forwarding to expose `katib-db-manager`, run this command:
+
+   ```
+   go run main.go --build-dir=../../../pkg/new-ui/v1beta1/frontend/dist --port=8080 --db-manager-address=localhost:6789
```

Comment:
Are you able to serve the UI from the `main.go` ?
I am getting `not found libs` error.

Reply:
Yes, where does this `not found libs` error appear? Is it on the request for index.html?

I'm able to see serve the app locally via the backend

https://user-images.githubusercontent.com/11134742/108832204-dcecfc80-75d3-11eb-991c-ba0e95df18bf.mp4

Reply:
Yes, with the updated script it is working. Thank you @kimwnasptd!

Btw, once I access to the UI the namespace is not selected (it might be because I don't have `kubeflow-user` namespace). Should we show All Namespaces by default or you think we should show the first one?
![no-namespace](https://user-images.githubusercontent.com/31112157/108848191-98b02b00-75d8-11eb-8fea-c5a296f4f8ee.gif)

Reply:
This is a good point. The reason you don't see a pre-selected value is because I've hardcoded to try and use the `kubeflow-user` namespace value. https://github.com/kubeflow/kubeflow/blob/0c4a738625f52092f66ec462e800ddd2837e0f6d/components/crud-web-apps/common/frontend/kubeflow-common-lib/projects/kubeflow/src/lib/services/namespace.service.ts#L38

This is going to be an issue, especially if the app should be deployed as standalone.
Indeed a good next step would be to select the first Namespace of the list. In the future we could do some more sophisticated things, like remembering the last Namespace the user had chosen and select it once the user navigates to the app again.

I'll add a comment to the general issue about standalone/kf mode. This should be tackled in the common code for all of the web apps https://github.com/kubeflow/kubeflow/issues/5566

Reply:
Make sense, thank you.

Reply:
I added a comment with next steps in both https://github.com/kubeflow/kubeflow/issues/5566#issuecomment-786576445 and in our katib issue as well https://github.com/kubeflow/katib/issues/1437#issuecomment-786579066 so we can keep track of this.

If you don't have any further suggestions for now then I think we can resolve this conversation.